rts/lra/lra-filters/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -394,26 +394,36 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
 
         try {
             if (current != null && isJaxRsCancel(requestContext, responseContext)) {
-                lraClient.cancelLRA(current);
-                if (current.toASCIIString().equals(
-                        Current.getLast(requestContext.getHeaders().get(LRA_HTTP_HEADER)))) {
-                    // the callers context was ended so invalidate it
-                    requestContext.getHeaders().remove(LRA_HTTP_HEADER);
-                }
+                try {
+                    lraClient.cancelLRA(current);
+                } catch (NotFoundException ignore) {
+                    // must already be cancelled
```

Comment:
I think it is useful to write a trace log in situation like this.

Reply:
If the LRA has already cancelled (either explicitly by the application or because it timed out then the exception path is normal behaviour and therefore does not need to be logged.

On the other hand if the application closed it then I would argue it is an application decision and therefore, again does not need to be logged.

Reply:
I agree, we should ensure that the spec is clear that cancelOn only applies when the transaction is still in a state it can be cancelled. I think the doc here should be updated to say:
// must already be completed

Reply:
I think there are cases when some error condition occurs and logging such situation to `trace` or `debug` is beneficial. Even it's assumed such situation it's expected or it should never happen.

Reply:
I do agree with the point that more verbose logging is useful typically I think the reason where this might not be appropriate to add in this case is for example we are not logging 
```
lraClient.cancelLRA(current);
traceln("Was cancelled);
```
In the same way Mike has analysed this and would say that the NotFoundException is expected in some cases.

Reply:
I would argue that exception is exception and it should be considered exceptional in any case. Still I'm fine the PR would be left as it is.